### PR TITLE
Deprecate `KerasLayer`

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
We're proposing to deprecate the KerasLayer class which has become increasingly difficult to maintain. During our [work to unpin NumPy](https://github.com/PennyLaneAI/pennylane/pull/7039) for compatibility with NumPy 2.x, we discovered that core part of `KerasLayer` has barely been updated since its initial implementation ~4 years ago and is now creating compatibility issues.

**Description of the Change:**
Mark `PennyLaneDeprecationWarning` in the `__init__` of `KerasLayer`

**Benefits:**
 - Less coupling with TF/Keras ecosys
 - More flexibility in the future for requriements adjustment

**Possible Drawbacks:**
 - Very small chance that there're still users that have been using PL+TF+Keras? It's hard to imagine though since `KerasLayer` simply does not support any Keras3

**Related GitHub Issues:**
[sc-86668]

### CODE REVIEW

File: pennylane/qnn/keras.py [hunk 1]
```diff
@@ -308,6 +311,11 @@ def __init__(
         weight_specs: Optional[dict] = None,
         **kwargs,
     ):
+        warnings.warn(
+            "The 'KerasLayer' class is deprecated and will be removed in v0.42. "
+            "This gradient recipe is not required for the new operator arithmetic system.",
```
Comment (albi3ro) [inline pennylane/qnn/keras.py:316]: Gradient recipe?

Comment (JerryChen97) [inline pennylane/qnn/keras.py:316]:
```suggestion
            "The 'KerasLayer' class is deprecated and will be removed in v0.42. ",
```

Comment (JerryChen97) [reply pennylane/qnn/keras.py]: Deleted

**Review by andrijapau — COMMENTED** [review]
I think it might be good to just add a sentence summarizing why we are deprecating it in the changelog / deprecations documentation. If I was a user I'd want to know. 😄

File: pennylane/qnn/keras.py [hunk 2]
```diff
@@ -14,11 +14,14 @@
 """This module contains the classes and functions for integrating QNodes with the Keras Layer
 API."""
 import inspect
+import warnings
 from collections.abc import Iterable
 from typing import Optional, Text
 
 from packaging.version import Version
 
+import pennylane as qml
```
Comment (andrijapau) [inline pennylane/qnn/keras.py:23]: I'm not sure how this works in detail, but don't we only need to import `PennyLaneDeprecationWarning`? Does this just import everything with it?

**Review by andrijapau — CHANGES_REQUESTED** [review]
Looks good! Just some suggestions about documentation. 😄

File: pennylane/qnn/keras.py (hunk 2 above)
Comment (JerryChen97) [reply pennylane/qnn/keras.py:23]: No nothing else! I think I better separately only import PLDW

Comment (JerryChen97) [reply pennylane/qnn/keras.py]: https://github.com/PennyLaneAI/pennylane/pull/7097/commits/19b9c6754e75d260013894beb4ba9c9106382a23

**Comment by andrijapau** [comment]
I'm thinking something like,
```
The `KerasLayer` in `qml.qnn.keras` is deprecated because Keras 2 is no longer actively maintained.  Please consider using a different machine learning framework instead of TensorFlow/Keras 2.
```
what do you think?

**Review by andrijapau — APPROVED** [review]
Awesome! LGTM 🚀

File: pennylane/qnn/keras.py [hunk 3]
```diff
@@ -51,6 +54,9 @@ class KerasLayer(Layer):
     `Model <https://www.tensorflow.org/api_docs/python/tf/keras/Model>`__ classes for
     creating quantum and hybrid models.
 
+    .. warning::
+        This class is deprecated because Keras 2 is no longer actively maintained.  Please consider using a different machine learning framework instead of TensorFlow/Keras 2.
```
Comment (albi3ro) [inline pennylane/qnn/keras.py:58]:
```suggestion
        This class is deprecated because Keras 2 is no longer actively maintained.  Please consider using torch instead of TensorFlow/Keras 2.
```
